libraries/app_helpers.rb

```diff
@@ -0,0 +1,90 @@
+
+# frozen_string_literal: true
+
+#
+# Cookbook Name:: cerner_splunk
+# File Name:: app_helpers.rb
+
+module CernerSplunk # rubocop:disable Style/Documentation
+  unless defined?(AppHelpers)
+    # Helper library for managing Splunk apps
+    module AppHelpers
+      def self.merge_hashes(*hashes)
+        hashes.collect(&:keys).flatten.uniq.each_with_object({}) do |app_name, result|
+          app_hash = {}
+
+          merger = proc { |_key, v1, v2| v1.is_a?(Hash) && v2.is_a?(Hash) ? v1.merge(v2, &merger) : v2 }
+
+          hashes.each do |hash|
+            to_merge = hash[app_name]
+            next unless to_merge.is_a? Hash
+            app_hash.merge!(to_merge, &merger)
+          end
+
+          result[app_name] = app_hash unless app_hash.empty?
+        end
+      end
+
+      def self.proc_files(**params)
+        files = (params[:files] || {}).select { |key, _| key.to_s =~ /.+(?<!\.conf)$/ }
+        lookups = params[:lookups] || {}
+
+        proc do |app_base|
+          app_path = Pathname.new app_base
+
+          files.each do |filename, contents|
+            file_path = app_path + filename
+
+            directory file_path.to_s do
+              owner node['splunk']['user']
+              group node['splunk']['group']
+              mode '0755'
+              recursive true
+            end
+
+            file file_path.to_s do
+              owner node['splunk']['user']
+              group node['splunk']['group']
+              mode '0600'
+              content CernerSplunk::ConfigProcs.parse(contents, filename: file_path)
+              action contents.empty? ? :delete : :create
+            end
+          end
+
+          lookups.each do |filename, url|
+            raise "Unsupported lookup file format for #{filename}" unless filename =~ /\.(?:csv\.gz|csv|kmz)$/i
```

Comment:
Future / unrelated thought: We probably want to log an issue about handling kmz properly. Splunk expands kmz lookups for caching/speed it seems and so manipulating the kmz file we may need to see if we also need to manipulate (delete) the directory too